File: Problem Solver/src/Problem010.cpp
#include "Problem010.h"
#include "Primes.h"
#include <iostream>
#include <vector>
#include <string>

using namespace std;

Problem010::Problem010()
{
    //ctor
}

Problem010::~Problem010()
{
    //dtor
}

void Problem010::Solve(){

    Primes prob;

    vector<unsigned long long> plist = prob.primeList(2000000);
    __int64 sum = 0;

    for (long long i = 0; i <= plist.size() - 1; i++){
        sum += plist.at(i);
    }

    string ans = to_string(sum);

    cout << ans << endl;
}

File: ProblemSolver/ProblemSolver/src/Problem010.cs
/*
Summation of primes
Problem 10
The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.
*/
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Numerics;

namespace ProblemSolver.src
{
    class Problem010 : SolvedProblems
    {
        public override void Solve()
        {
            Dictionary<BigInteger, BigInteger> plist = Primes.primeList(BigInteger.Parse("2000000"));
            ulong sum = 0;

            for (int i = 1; i < plist.Count; i++)
            {
                sum += (ulong)plist[i];
            }

            EulerHelper.printAns(sum.ToString());
        }
    }
}
